### ISSUE TITLE
Error reading from OrientDB via Gremlin Server

### ISSUE THREAD

**Issue by amachwe** [orientechnologies/orientdb#8246]
### Java Version: openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-8u162-b12-0ubuntu0.17.10.2-b12)
OpenJDK 64-Bit Server VM (build 25.162-b12, mixed mode)

### OS: Ubuntu 17.10
Gremlin Server 3.3.2
## Expected behavior  
Read value from Orient DB via Gremlin

## Actual behavior  
[WARN] AbstractGraphSONMessageSerializerV2d0 - Response [ResponseMessage{requestId=0f60a51d-9d30-4a8f-9074-aa174a30a2a5, status=ResponseStatus{code=SUCCESS, message='', attributes={}}, result=ResponseResult{data=[v[#15:17]], meta={}}}] could not be serialized by org.apache.tinkerpop.gremlin.driver.ser.AbstractGraphSONMessageSerializerV2d0.
org.apache.tinkerpop.shaded.jackson.databind.exc.InvalidDefinitionException: Direct self-reference leading to cycle (through reference chain: com.orientechnologies.orient.core.id.ORecordId["identity"])
	at org.apache.tinkerpop.shaded.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
.......
Caused by: org.apache.tinkerpop.shaded.jackson.databind.exc.InvalidDefinitionException: Direct self-reference leading to cycle (through reference chain: com.orientechnologies.orient.core.id.ORecordId["identity"])

Error seen in python client:
GremlinServerError: 599: Error during serialization: Direct self-reference leading to cycle (through reference chain: com.orientechnologies.orient.core.id.ORecordId["record"]->com.orientechnologies.orient.core.record.impl.ODocument["schemaClass"]->com.orientechnologies.orient.core.metadata.schema.OClassImpl["document"]->com.orientechnologies.orient.core.record.impl.ODocument["owners"]->com.orientechnologies.orient.core.record.impl.ODocument["record"])

## Steps to reproduce  

Using GremlinPython

g = graph.traversal().withRemote(DriverRemoteConnection('ws://localhost:8182/gremlin','g'))
 g.addV().property("node",str(node)).property("node_id",str(node_id)).property("type","node").toList()

>> In above - node is added in orientdb but exception is thrown. If you want to create edges it is impossible to do as as one is not able to read the right vertex from orientdb.

**Comment by Akshay-15mca1098** [comment]
I am also dealing with the same problem, did you solved it

**Comment by amachwe** [comment]
No, we were forced to move to Neo4j

**Comment by maggiolo00** [comment]
Hi @amachwe 

sorry to hear that. From OrientDB 3.0 we started to ship the official support of Apache Tinkerpop

We ship also a version with GremlinServer embedded that you can download here

https://orientdb.com/download-2/

**Comment by amachwe** [comment]
Hello Enrico,
Yes we did try it recently with the embedded gremlin but it seems that
there are still issues.

Let me explain my use case.
We have a gremlin Python app that writes to gremlin server - we want to
persist the graph and also query it via other apps directly - and we found
the orient db browser very useful.

We had remote orient db connected to gremlin server (standalone) which our
app was talking to (via gremlin Python).

It would also be nice if you guys update the tinker pop page where they
have sections for different ports like neo4j gremlin and Hadoop gremlin.

Thank you,
Azahar
